tls/s2n_connection.c

```diff
@@ -1033,11 +1033,13 @@ const char *s2n_connection_get_curve(struct s2n_connection *conn)
 {
     PTR_ENSURE_REF(conn);
 
-    if (!conn->secure.server_ecc_evp_params.negotiated_curve) {
-        return "NONE";
+    if (conn->secure.server_ecc_evp_params.negotiated_curve && 
+        (s2n_kex_includes(conn->secure.cipher_suite->key_exchange_alg, &s2n_ecdhe) ||
+         s2n_kex_includes(conn->secure.cipher_suite->key_exchange_alg, &s2n_hybrid_ecdhe_kem))) {
```

Comment:
this if clause is getting harder to read. can we pull this out into a separate function. maybe `negotiated_ecdhe_cipher_suite`

Reply:
Well, I realized we don't need one of the cases here, so lmk if this revision still needs a separate function.